internal/application/command.go

```diff
@@ -138,6 +140,62 @@ func readDeviceResource(device models.Device, resourceName string, attributes st
 	return res, nil
 }
 
+func readDeviceResources(device models.Device, regexResourceName string, attributes string, dic *di.Container) (res *dtos.Event, edgexErr errors.EdgeX) {
+	deviceResources, ok := cache.Profiles().DeviceResourcesByRegex(device.ProfileName, regexResourceName)
+	if !ok || len(deviceResources) == 0 {
+		errMsg := fmt.Sprintf("Regex deviceResource %s not found", regexResourceName)
+		return res, errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, errMsg, nil)
+	}
+
+	var woCounter int
+	lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+	reqs := make([]sdkModels.CommandRequest, 0)
+	for _, dr := range deviceResources {
+		// check deviceResource is not write-only
+		if dr.Properties.ReadWrite == common.ReadWrite_W {
+			lc.Debugf("deviceResource %s is marked as write-only", dr.Name)
+			woCounter++
+			continue
+		}
+
+		// prepare CommandRequest
+		var req sdkModels.CommandRequest
+		req.DeviceResourceName = dr.Name
+		req.Attributes = dr.Attributes
+		if attributes != "" {
+			if len(req.Attributes) <= 0 {
+				req.Attributes = make(map[string]any)
+			}
+			req.Attributes[sdkCommon.URLRawQuery] = attributes
+		}
+		req.Type = dr.Properties.ValueType
+
+		reqs = append(reqs, req)
+	}
+
+	if woCounter == len(deviceResources) {
+		errMsg := fmt.Sprintf("all deviceResources matched with %s are marked as write-only", regexResourceName)
+		return res, errors.NewCommonEdgeX(errors.KindNotAllowed, errMsg, nil)
+	}
```

Comment:
Can't we just check `len(reqs)` and not have `woCounter `?
```suggestion
	if  len(reqs ) {
		errMsg := fmt.Sprintf("no readable resources matched with %s", regexResourceName)
		return res, errors.NewCommonEdgeX(errors.KindNotAllowed, errMsg, nil)
	}
```